fre/make/runFremake.py

```diff
@@ -56,136 +61,146 @@ def fremake_run(yamlfile,platform,target,parallel,jobs,no_parallel_checkout,verb
 
     ## Error checking the targets
     for targetName in tlist:
-         target = targetfre.fretarget(targetName)
+        target = targetfre.fretarget(targetName)
 
     ## Loop through the platforms specified on the command line
     ## If the platform is a baremetal platform, write the checkout script and run it once
     ## This should be done separately and serially because bare metal platforms should all be using
     ## the same source code.
     for platformName in plist:
-         if modelYaml.platforms.hasPlatform(platformName):
-              pass
-         else:
-              raise ValueError (platformName + " does not exist in " + modelYaml.combined.get("compile").get("platformYaml"))
-
-         (compiler,modules,modulesInit,fc,cc,modelRoot,iscontainer,mkTemplate,containerBuild,ContainerRun,RUNenv)=modelYaml.platforms.getPlatformFromName(platformName)
-
-         ## Create the checkout script
-         if iscontainer == False:
-              ## Create the source directory for the platform
-              srcDir = modelRoot + "/" + fremakeYaml["experiment"] + "/src"
-              if not os.path.exists(srcDir):
-                   os.system("mkdir -p " + srcDir)
-              if not os.path.exists(srcDir+"/checkout.sh"):
-                   freCheckout = checkout.checkout("checkout.sh",srcDir)
-                   freCheckout.writeCheckout(modelYaml.compile.getCompileYaml(),jobs,pc)
-                   freCheckout.finish(pc)
-
-    ## TODO: Options for running on login cluster?
-                   freCheckout.run()
+        if modelYaml.platforms.hasPlatform(platformName):
+            pass
+        else:
+            raise ValueError(f'{platformName} does not exist in '
+                             f'{modelYaml.combined.get("compile").get("platformYaml")}')
+
+        ( compiler, modules, modulesInit,
```

Comment:
Thank you for fixing this some more!

Reply:
f-strings are your friend and so are built-in exceptions!